data/multiplayer/scenarios/2p_Isle_of_Mists.lua

```diff
@@ -0,0 +1,534 @@
+-- The weather events are completely unrelated to the spawn events.
+
+local function get_weather_duration(max_duration)
+	local res = wesnoth.random(2)
+	while res < max_duration and wesnoth.random(2) == 2 do
+		res = res + 1
+	end
+	return res
+end
+-- initilize the weather_event wml array which defines at which turns the weather changes.
+on_event("prestart", function()
+	local turn = wesnoth.random(4,6)
+	local event_num = 0
+	local weather_to_dispense = {
+		{ turns_left = 10, id = "drought"},
+		{ turns_left = 12, id = "heavy rain"},
+		{ turns_left = 9, id = "snowfall"},
+	}
+	local clear_turns_left = 22
+	local heavy_snowfall_turns_left = 6
+	local second_rain_turns_left = 12
+	while turn < 55 and #weather_to_dispense > 0 do
+		-- pick a random weather except 'clear' and 'heavy snow'
+		local index = wesnoth.random(#weather_to_dispense)
+		local num_turns = get_weather_duration(weather_to_dispense[index].turns_left)
+		local weather_id = weather_to_dispense[index].id
+		weather_to_dispense[index].turns_left = weather_to_dispense[index].turns_left - num_turns
+		if weather_to_dispense[index].turns_left <= 0 then
+			table.remove(weather_to_dispense, index)
+		end
+		wml.variables[string.format("weather_event[%d]", event_num)] = {
+			turn = turn,
+			weather_id = weather_id,
+		}
+		event_num = event_num + 1
+		turn = turn + num_turns
+		-- Second snow happens half the time.
+		if weather_id == "snowfall" and heavy_snowfall_turns_left >= 0 and wesnoth.random(2) == 2 then
+			num_turns = get_weather_duration(heavy_snowfall_turns_left)
+			wml.variables[string.format("weather_event[%d]", event_num)] = {
+				turn = turn,
+				weather_id = "heavy snowfall",
+			}
+			event_num = event_num + 1
+			turn = turn + num_turns
+			heavy_snowfall_turns_left = heavy_snowfall_turns_left - num_turns
+		end
+
+		-- second heavy rain/inundation happens one-thirds the time
+		if weather_id == "heavy rain" and second_rain_turns_left >= 0 and wesnoth.random(3) == 3 then
+			num_turns = get_weather_duration(second_rain_turns_left)
+			wml.variables[string.format("weather_event[%d]", event_num)] = {
+				turn = turn,
+				weather_id = "second rain",
+			}
+			event_num = event_num + 1
+			turn = turn + num_turns
+			second_rain_turns_left = second_rain_turns_left - num_turns
+		end
+
+		-- Go back to clear weather.
+		num_turns = get_weather_duration(clear_turns_left)
+		wml.variables[string.format("weather_event[%d]", event_num)] = {
+			turn = turn,
+			weather_id = "clear",
+		}
+		event_num = event_num + 1
+		turn = turn + num_turns
+		clear_turns_left = clear_turns_left - num_turns
+	end
+end)
+
+local function weather_alert(text, red, green, blue)
+	wesnoth.wml_actions.print {
+		text = text,
+		duration = 120,
+		size = 26,
+		red = red,
+		green = green,
+		blue = blue,
+	}
+end
+
+local function weather_map(name)
+	wesnoth.wml_actions.terrain_mask {
+		mask = wesnoth.read_file(name),
```

Comment:
This file will need the same changes that Dark Forecast received in PR #5537 .

Reply:
Seems simple enough. Nothing mass-search and replace cannot remedy. I shall push 2 commits with fixes. One for this, the other for the "balance".

Reply:
I think I patched all of them.